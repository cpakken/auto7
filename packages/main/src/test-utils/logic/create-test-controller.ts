import { objectToArray } from "@utils/collection-fns"
import { LogicController, ILogicController, LogicNativeSnapshotIn, LogicComposedSnapshotIn } from "@main/controllers/logic"
import { createLogicNativeStoreSnapshot, createLogicComposedStoreSnapshot } from "./snapshot-factories"

import * as nativeSnapshots from "./native/snapshots"
import * as composedSnapshots from "./composed"

export function createTestLogicController(env?): ILogicController {
  const nativeLoaders: LogicNativeSnapshotIn[] = objectToArray(nativeSnapshots)
  const native = createLogicNativeStoreSnapshot(nativeLoaders)

  const composedLoaders: LogicComposedSnapshotIn[] = objectToArray(composedSnapshots)
  const composed = createLogicComposedStoreSnapshot(composedLoaders)

  const snapshot = { composed, native }

  //TODO process cachedKeys
  return LogicController.create(snapshot, env)
}
