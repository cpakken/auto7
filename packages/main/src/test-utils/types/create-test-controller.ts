import { objectToArray } from "@utils/collection-fns"
import { ITypeController, TypeController, TypeNativeSnapshotIn } from "@main/controllers/types"
import { createTypeNativeStoreSnapshot } from "./snapshot-factories"

import * as nativeSnapshots from "./native/std.config"

export function createTestTypeController(env?): ITypeController {
  const nativeLoaders: TypeNativeSnapshotIn[] = objectToArray(nativeSnapshots)
  const native = createTypeNativeStoreSnapshot(nativeLoaders)

  // const composedLoaders: LogicComposedSnapshotIn[] = objectToArray(composedSnapshots)
  // const composed = createLogicComposedStoreSnapshot(composedLoaders)

  const snapshot = {
    native,
    // composed,
  }

  //TODO process cachedKeys
  return TypeController.create(snapshot, env)
}
