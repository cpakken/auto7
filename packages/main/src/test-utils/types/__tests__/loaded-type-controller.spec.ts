import { getSnapshot } from "mobx-state-tree"
import { createTestTypeController } from "../create-test-controller"

describe("loadedTypeController", () => {
  test("inits", () => {
    const controller = createTestTypeController()
    // const native = controller.native.get("_std_string")
    expect(getSnapshot(controller)).toMatchSnapshot()
  })
})
