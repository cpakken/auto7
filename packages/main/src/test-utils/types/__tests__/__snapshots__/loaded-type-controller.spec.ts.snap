// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadedTypeController inits 1`] = `
Object {
  "composed": Object {
    "composition": Array [],
    "info": Array [],
  },
  "native": Object {
    "info": Array [
      Object {
        "avatar": Object {
          "abbr": "abc",
          "color": "rose",
        },
        "description": "A string of word/characters/number etc.",
        "label": "string",
      },
      Object {
        "avatar": Object {
          "color": "teal",
          "icon": "number",
        },
        "description": "A number to perform mathematic operations",
        "label": "number",
      },
    ],
  },
}
`;
